src/update_roster.R

```diff
@@ -1,642 +0,0 @@
-        entry_year,
-        rookie_year,
-        draft_club,
-        gsis_it_id,
-        smart_id,
-        height,
-        weight
-      ) |>
-      dplyr::mutate(birth_date = as.Date(birth_date), dplyr::across(c(gsis_it_id, smart_id), as.character))
-
-    if(season < 2016)
-      shield <- build_rosters_shieldapi(season) |> fill_ids()
-
-    if (season < 2002)
-      roster <- shield
-
-    if (season >= 2002 && season < 2016) {
-      weekly_rosters <- build_rosters_weekly_dataexchange(season)
-      roster <- convert_weekly_to_season_rosters(weekly_rosters)
-      weekly_rosters <- weekly_rosters |>
-        dplyr::select(-c(status, height, weight, headshot)) |>
-        dplyr::left_join(
-          shield |>
-            dplyr::select(gsis_id, status, height, weight, headshot = headshot_url, birth_date) |>
-            dplyr::arrange(gsis_id, status) |>
-            dplyr::distinct(.keep_all = TRUE),
-          by = c("gsis_id"),
-          na_matches = "never"
-        ) |>
-        fill_ids()
-
-      roster <- roster |>
-        dplyr::select(-c(status, height, weight, headshot)) |>
-        dplyr::left_join(
-          shield |>
-            dplyr::select(gsis_id, status, height, weight, headshot = headshot_url, birth_date) |>
-            dplyr::arrange(gsis_id, status) |>
-            dplyr::distinct(gsis_id, .keep_all = TRUE),
-          by = c("gsis_id"),
-          na_matches = "never"
-        ) |>
-        fill_ids()
-    }
-
-    if (season >= 2016) {
-      weekly_rosters <- purrr::possibly(build_rosters_weekly_ngsapi, data.frame())(season)
-      if (nrow(weekly_rosters) > 0) {
-        roster <- convert_weekly_to_season_rosters(weekly_rosters)
-        shield <- build_rosters_shieldapi(season)
-        weekly_rosters <- weekly_rosters |>
-          dplyr::select(-c(height)) |>
-          dplyr::left_join(
-            shield |>
-              dplyr::select(gsis_id, birth_date, height, headshot_gsis = headshot) |>
-              dplyr::distinct(),
-            by = c("gsis_id"),
-            na_matches = "never"
-          ) |>
-          dplyr::mutate(headshot = dplyr::coalesce(headshot, headshot_gsis)) |>
-          dplyr::select(-c(headshot_gsis)) |>
-          fill_ids()
-
-        roster <- roster |>
-          dplyr::select(-c(height)) |>
-          dplyr::left_join(
-            shield |>
-              dplyr::select(gsis_id, birth_date, height, headshot_gsis = headshot) |>
-              dplyr::distinct(),
-            by = c("gsis_id"),
-            na_matches = "never"
-          ) |>
-          dplyr::mutate(headshot = dplyr::coalesce(headshot, headshot_gsis)) |>
-          dplyr::select(-c(headshot_gsis)) |>
-          fill_ids()
-      }
-    }
-
-    if (season >= 2016 && nrow(weekly_rosters) == 0) {
-      roster <- shield
-    }
-
-    if ((season < 2002 && length(roster) == 0) |
-        (season >= 2002 && length(weekly_rosters) == 0)) {
-      return(invisible(NULL))
-    }
```

Comment:
I think this is the reason the roster jobs were "failing" - it would return invisibly if there were no weekly rosters here but we do want to continue, so I updated it to check for nrow(weekly_rosters) instead